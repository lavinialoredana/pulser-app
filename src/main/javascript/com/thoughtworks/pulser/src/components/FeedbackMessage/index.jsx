import React from "react";

const FeedbackMessage = ({ userFeedback, onUserFeedbackChange }) => {
  return (
    <div>
      <textarea
        data-testid="textarea-field"
        rows="4"
        cols="80"
        placeholder="Write here any additional feedback"
        value={userFeedback}
        onChange={onUserFeedbackChange}
      ></textarea>
    </div>
  );
};

export default FeedbackMessage;
