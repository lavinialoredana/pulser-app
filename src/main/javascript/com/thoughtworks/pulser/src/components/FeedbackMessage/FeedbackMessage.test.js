import { cleanup, render, screen } from "@testing-library/react";
import FeedbackMessage from ".";
import userEvent from "@testing-library/user-event";

describe("FeedbackMessage component", () => {
  afterEach(() => {
    cleanup();
  });

  it("should match snapshot", ()=>{
      const {asFragment} = render(<FeedbackMessage/>);

      expect(asFragment()).toMatchSnapshot();
  })

  it("should contain textarea field", () => {
    render(<FeedbackMessage />);

    const inputFieldElement = screen.getByTestId("textarea-field");

    expect(inputFieldElement).toBeInTheDocument();
  });

  it("should contain placeholder", () => {
    render(<FeedbackMessage />);

    const placeholderText = screen.getByPlaceholderText(
      "Write here any additional feedback"
    );

    expect(placeholderText).toBeInTheDocument();
  });

  it("should get value from textarea", () => {
    const message = "Thank you!";

    render(<FeedbackMessage userFeedback={message} />);

    const userInput = screen.getByTestId("textarea-field");

    expect(userInput.value).toBe(message);
  });

  it("should fire onChange correctly", async () => {
    const user = userEvent.setup();
    const message = "Thank you!";
    const onChangeMock = jest.fn();

    render(<FeedbackMessage onUserFeedbackChange={onChangeMock} />);

    const userInput = screen.getByTestId("textarea-field");

    await user.type(userInput, message);
    expect(onChangeMock).toHaveBeenCalledTimes(message.length);
  });
});
