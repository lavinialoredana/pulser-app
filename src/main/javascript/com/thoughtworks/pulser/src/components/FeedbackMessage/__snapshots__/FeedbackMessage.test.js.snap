// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackMessage component should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <textarea
      cols="80"
      data-testid="textarea-field"
      placeholder="Write here any additional feedback"
      rows="4"
    />
  </div>
</DocumentFragment>
`;
